redis_cache.go

```diff
@@ -0,0 +1,134 @@
+package lcw
+
+import (
+	"sync/atomic"
+	"time"
+
+	redis "github.com/go-redis/redis/v7"
+	"github.com/pkg/errors"
+)
+
+// RedisSizeLimit is maximum allowed value size in Redis
+const RedisSizeLimit = 512 * 1024 * 1024
+
+// RedisCache implements LoadingCache for Redis.
+type RedisCache struct {
+	options
+	CacheStat
+	backend *redis.Client
+}
+
+// NewRedisCache makes Redis LoadingCache implementation.
+func NewRedisCache(backend *redis.Client, opts ...Option) (*RedisCache, error) {
+
+	res := RedisCache{
+		options: options{
+			ttl: 5 * time.Minute,
+		},
+	}
+	for _, opt := range opts {
+		if err := opt(&res.options); err != nil {
+			return nil, errors.Wrap(err, "failed to set cache option")
+		}
+	}
+
+	res.backend = backend
+
+	return &res, nil
+}
+
+// Get gets value by key or load with fn if not found in cache
+func (c *RedisCache) Get(key string, fn func() (Value, error)) (data Value, err error) {
+
+	v, getErr := c.backend.Get(key).Result()
+	switch getErr {
+	case nil:
```

Comment:
those cases better to be commented a little bit, because it takes time to figure what nil vs redis.Nil supposed to mean